Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using Modix.Common.Messaging;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Promotions;
+using Modix.Services.Core;
+using Modix.Services.Images;
+using Modix.Services.Promotions;
+using Modix.Services.Utilities;
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogueBehavior :
+        INotificationHandler<ReactionAddedNotification>,
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly DateTime _utcNow = DateTime.UtcNow;
+
+        private static readonly Emoji _approveEmoji = new Emoji("👍");
+
+        private static readonly Emoji _disproveEmoji = new Emoji("👎");
+
+        private IDesignatedChannelService _designatedChannelService { get; }
+
+        private IDiscordSocketClient _discordSocketClient { get; }
+
+        private IMemoryCache _memoryCache { get; }
+
+        private IPromotionsService _promotionsService { get; }
+
+        private IImageService _imageService { get; }
+
+        public PromotionDialogueBehavior(IDesignatedChannelService designatedChannelService,
```

Comment:
Inject an `ILogger<PromotionDialogueBehavior>` and do some logging.

Reply:
Any particular thing to be logged?

Reply:
EVERYTHING

Ideally, every decision point should be at least a TRACE log. You're free to experiment a bit here, we don't have a lot of "first-class" logging elsewhere in the app.